### DISCUSSION TITLE
Body exceeded 50000kb limit

### DISCUSSION THREAD

**Discussion by DannielSouza** [vercel/community#4836] [Help]
### Summary

Hey everyone! I saw that some people are having problems when their Serverless Function exceeds 50mb. I'm having a similar error, but it doesn't look like that the error is in a Serveless Function, but in the body. Initially I was using Next11 when I had this problem, I upgraded to Next13 and I'm still getting the same, is this related to me using Firebase?

The error in the console says:

```
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
11:21:14.825 | (ISR)     incremental static regeneration (uses revalidate in getStaticProps)
11:21:14.825 |  
11:21:14.890 | Done in 596.71s.
11:21:44.464 | Traced Next.js server files in: 395.591ms
11:21:49.592 | Created all serverless functions in: 5.126s
11:21:51.613 | Collected static files (public/, static/, .next/static): 35.888ms
11:22:31.425 | Build Completed in /vercel/output [12m]
11:23:13.903 | Deploying outputs...
11:38:56.387 | Error: Body exceeded 50000kb limit
```

### Steps to Reproduce

- Just trying to do a regular deploy

**Comment by Legym** [comment]
@DannielSouza  ive spent hours trying to understand why im getting this error. Did you find a solution?

**Reply by Legym** [reply]
After digging though more docs, i found this article. It appears that you can only SSG up to 15k. The solution is to generate up to 15k of your most important pages then use ISR for the rest

https://vercel.com/docs/limits/overview#files

**Comment by amyegan** [comment]
This error can happen if the build attempts to pre-render too many pages. As @Legym mentioned, [Incremental Static Regeneration (ISR)](https://vercel.com/docs/incremental-static-regeneration#) can be used to move the rendering of some pages to the server side. 

Another option is to run the [build locally using Vercel CLI](https://vercel.com/docs/cli/deploying-from-cli#deploying-from-local-build-prebuilt). You can deploy the prebuilt output using [`vercel deploy --prebuilt`](https://vercel.com/docs/cli/deploy#). If you run into the 15,000 source file limit with that, then you can add the [`--archive=tgz`](https://vercel.com/docs/cli/deploy#archive) option to compress the deployment code into a single file before uploading it.
